ocaml/typing/typecore.ml

```diff
@@ -7406,7 +7460,7 @@ and type_argument ?explanation ?recarg env (mode : expected_mode) sarg
          Texp_ident(Path.Pident id, mknoloc (Longident.Lident name),
                     desc, Id_value, uu)}
       in
-      let eta_mode = Value.local_to_regional (Value.of_alloc marg) in
+      let eta_mode, _ = Value.newvar_below (alloc_as_value marg) in
```

Comment:
Why the change here?

Reply:
This was probably a bug introduced when refactoring.

I just spent quite some time to produce a test case to prevent this in the future, but couldn't do so, because the `eta_mode` is used quite locally (only used in a `Tpat_var` and the corresponding `Texp_ident`). I'm not sure if this bug is observable. For now I will just fix this.